hid_darwin.go

```diff
@@ -213,12 +213,17 @@ func getPath(osDev C.IOHIDDeviceRef) string {
 }
 
 func iterateDevices(action func(device C.IOHIDDeviceRef) bool) cleanupDeviceManagerFn {
-	mgr := C.IOHIDManagerCreate(C.kCFAllocatorDefault, C.kIOHIDOptionsTypeNone)
+	var mgr C.IOHIDManagerRef
+	mgr = C.IOHIDManagerCreate(C.kCFAllocatorDefault, C.kIOHIDOptionsTypeNone)
 	C.IOHIDManagerSetDeviceMatching(mgr, nil)
 	C.IOHIDManagerOpen(mgr, C.kIOHIDOptionsTypeNone)
 
-	allDevicesSet := C.IOHIDManagerCopyDevices(mgr)
-	defer C.CFRelease(C.CFTypeRef(allDevicesSet))
+	var allDevicesSet C.CFSetRef
+	allDevicesSet = C.IOHIDManagerCopyDevices(mgr)
```

Comment:
`IOHIDManagerCopyDevices` can return nil, so we should check for that here to avoid an issue when we release it later.